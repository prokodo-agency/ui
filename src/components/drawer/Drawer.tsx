import { createIsland } from "@/helpers/createIsland"

import DrawerServer from "./Drawer.server"

import type { DrawerProps } from "./Drawer.model"

/**
 * The “Island” entrypoint.  Consumers import this file:
 *
 *    import { Drawer } from '@/components/drawer'
 *
 * and use <Drawer open={...} ...>...</Drawer> in their JSX.
 */
export const Drawer = createIsland<DrawerProps>({
  name: "Drawer",
  Server: DrawerServer,
  loadLazy: () => import("./Drawer.lazy"),
})

export default Drawer
