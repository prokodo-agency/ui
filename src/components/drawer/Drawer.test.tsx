jest.mock("@/components/icon/getIconLoader", () => ({
  getIconLoader: () => null
}))

import { expect } from "@jest/globals"

import { render } from "@/tests"

import { Drawer } from "./Drawer"

jest.mock(
  "remark-breaks",
  () =>
    ({ children }) =>
      children,
)
jest.mock(
  "remark-gfm",
  () =>
    ({ children }) =>
      children,
)

describe("The common drawer component", () => {
  it("should render drawer", async () => {
    const { container } = render(
      <Drawer anchor="right" open={true} onClose={jest.fn()} />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
