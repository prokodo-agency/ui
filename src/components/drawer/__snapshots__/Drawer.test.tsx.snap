// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The common drawer component should render drawer 1`] = `
<div
  data-island="drawer"
/>
`;
